src/material-experimental/mdc-slider/slider.ts

```diff
@@ -87,214 +65,264 @@ export class MatSliderChange {
-    } else {
-      this._tickInterval = 0;
-    }
+  @Input() displayWith: ((value: number) => string) | null;
+
+  /** Whether this is a ranged slider. */
+  get isRange(): boolean { return this.inputs.length === 2; }
+
+  tickMarks: TickMark[] = [];
+
+  initialized: boolean = false;
+
+  private _getWindow() {
+    return this._document.defaultView || window;
   }
-  private _tickInterval: number|'auto' = 0;
 
-  /** Whether or not to show the thumb label. */
-  @Input()
-  get thumbLabel(): boolean {
-    return this._thumbLabel;
+  private _getThumb(thumb: Thumb): MatSliderThumb {
+    return thumb === Thumb.END ? this.thumbs[this.thumbs.length - 1] : this.thumbs[0];
   }
-  set thumbLabel(value: boolean) {
-    this._thumbLabel = coerceBooleanProperty(value);
+
+  private _getThumbEl(thumb: Thumb): HTMLElement {
+    return this._getThumb(thumb).getRootEl();
   }
-  private _thumbLabel: boolean = false;
 
-  /** Whether the slider is disabled. */
-  @Input()
-  get disabled(): boolean {
-    return this._disabled;
+  private _getInput(thumb: Thumb): MatSliderInput {
+    return thumb === Thumb.END ? this.inputs[this.inputs.length - 1] : this.inputs[0];
   }
-  set disabled(disabled) {
-    this._disabled = coerceBooleanProperty(disabled);
+
+  private _getInputEl(thumb: Thumb): HTMLInputElement {
+    return this._getInput(thumb).getRootEl();
   }
-  private _disabled = false;
 
   /** Adapter for the MDC slider foundation. */
   private _sliderAdapter: MDCSliderAdapter = {
-    hasClass: (_className: string) => false,
-    addClass: (_className: string) => {},
-    removeClass: (_className: string) => {},
-    getAttribute: (_attribute: string) => null,
-    addThumbClass: (_className: string, _thumb: Thumb) => {},
-    removeThumbClass: (_className: string, _thumb: Thumb) => {},
-    getThumbKnobWidth: (_thumb: Thumb) => 0,
-    getThumbBoundingClientRect: (_thumb: Thumb) => null!,
-    getBoundingClientRect: () => null!,
-    isRTL: () => false,
-    setThumbStyleProperty: (_propertyName: string, _value: string, _thumb: Thumb) => {},
-    removeThumbStyleProperty: (_propertyName: string, _thumb: Thumb) => {},
-    setTrackActiveStyleProperty: (_propertyName: string, _value: string) => {},
-    setValueIndicatorText: (_value: number, _thumb: Thumb) => {},
-    updateTickMarks: () => {},
-    setPointerCapture: (_pointerId: number) => {},
-    emitChangeEvent: (_value: number, _thumb: Thumb) => {},
-    emitInputEvent: (_value: number, _thumb: Thumb) => {},
-    registerEventHandler: () => {},
-    deregisterEventHandler: () => {},
-    registerThumbEventHandler: () => {},
-    deregisterThumbEventHandler: () => {},
-    registerBodyEventHandler: () => {},
-    deregisterBodyEventHandler: () => {},
-    registerWindowEventHandler: () => {},
-    deregisterWindowEventHandler: () => {},
-    removeTrackActiveStyleProperty: (_propertyName: string) => {},
+    hasClass: (_className: string) => {
```

Comment:
Can you extract this adapter out into a standalone class? See this PR for a reference https://github.com/angular/components/pull/19980